ui/app/components/transit-key-actions.js

```diff
@@ -35,213 +50,142 @@ const TRANSIT_PARAMS = {
-  },
-
-  didReceiveAttrs() {
-    this._super(...arguments);
-    this.checkAction();
-    if (this.selectedAction === 'export') {
-      this.setExportKeyDefaults();
-    }
-  },
-
-  setExportKeyDefaults() {
-    const exportKeyType = this.key.exportKeyTypes.firstObject;
-    const exportKeyVersion = this.key.validKeyVersions.lastObject;
-    this.setProperties({
-      exportKeyType,
-      exportKeyVersion,
-    });
-  },
-
-  keyIsRSA: computed('key.type', function () {
-    const type = this.key.type;
-    return type === 'rsa-2048' || type === 'rsa-3072' || type === 'rsa-4096';
-  }),
 
-  getModelInfo() {
-    const model = this.key || this.backend;
-    if (!model) {
-      return null;
-    }
-    const backend = model.backend || model.id;
-    const id = model.id;
-
-    return {
-      backend,
-      id,
-    };
-  },
-
-  checkAction() {
-    const currentAction = this.selectedAction;
-    const oldAction = this.oldSelectedAction;
-
-    this.resetParams(oldAction, currentAction);
-    set(this, 'oldSelectedAction', currentAction);
-  },
-
-  resetParams(oldAction, action) {
-    const params = copy(TRANSIT_PARAMS);
-    let paramsToKeep;
-    const clearWithoutCheck =
-      !oldAction ||
-      // don't save values from datakey
-      oldAction === 'datakey' ||
-      // can rewrap signatures — using that as a ciphertext later would be problematic
-      (oldAction === 'rewrap' && !this.key.supportsEncryption);
-
-    if (!clearWithoutCheck && action) {
-      paramsToKeep = PARAMS_FOR_ACTION[action];
-    }
+export default class TransitKeyActions extends Component {
+  @service store;
+  @service flashMessages;
+  @service router;
 
-    if (paramsToKeep) {
-      paramsToKeep.forEach((param) => delete params[param]);
-    }
-    //resets params still left in the object to defaults
-    this.clearErrors();
-    this.setProperties(params);
-    if (action === 'export') {
-      this.setExportKeyDefaults();
-    }
-  },
+  @tracked isModalActive = false;
+  @tracked errors = null;
+  @tracked props = { ...STARTING_TRANSIT_PROPS }; // Shallow copy of the object. We don't want to mutate the original.
 
-  handleError(e) {
-    this.set('errors', e.errors);
-  },
+  constructor() {
+    super(...arguments);
+    assert('@key is required for TransitKeyActions components', this.args.key);
 
-  clearErrors() {
-    this.set('errors', null);
-  },
+    if (this.firstSupportedAction === 'export' || this.args.selectedAction === 'export') {
+      this.props.exportKeyType = this.args.key.exportKeyTypes[0];
+      this.props.exportKeyVersion = this.args.key.validKeyVersions.lastObject;
+    }
+  }
+
+  get keyIsRSA() {
+    const { type } = this.args.key;
+    return type === 'rsa-2048' || type === 'rsa-3072' || type === 'rsa-4096';
+  }
 
-  triggerSuccessMessage(action) {
-    const message = SUCCESS_MESSAGE_FOR_ACTION[action];
-    if (!message) return;
-    this.flashMessages.success(message);
-  },
+  get firstSupportedAction() {
```

Comment:
Should this just return `keyAction`? 
```js
 return this.args.selectedAction || this.args.key.supportedActions[0]
```
Then above could just check `this.keyAction === 'export'` and below on 169 it can be `const action = this.keyAction`

Reply:
I'm not sure I follow the change here. "Export" is a unique situation where it's not always returned as a queryParam and thus this.args.selectedAction === ''. So we confirm that it's indeed "export" by checking the model (called "key") and store the value as a tracked property in case it changes. We then check for it on the constructor and submit. The pattern here follows as close as made sense the original flow: check for the passed in selectedAction, if it didn't exists, then return it from the firstObject of the model passed in.

```
#from original init function
// eslint-disable-next-line ember/no-get
    if (this.selectedAction) {
      return;
    }
    // eslint-disable-next-line ember/no-get
    set(this, 'selectedAction', get(this, 'key.supportedActions.firstObject'));
```

Reply:
I was suggesting updating the getter to be clearer